src/runners/scilla_runner.ml

```diff
@@ -141,12 +141,26 @@ let deploy_library (cli : Cli.ioFiles) gas_remaining =
         | Invalid_json s ->
             fatal_error_gas (s @ (mk_error0 (sprintf "Failed to parse json %s:\n" cli.input_init))) gas_remaining'
       in
-      (* init.json for libraries can only have _extlibs field. *)
-      (match initargs with
-      | [(label, _)] when label = extlibs_label -> ()
-      | _ -> perr @@ scilla_error_gas_string gas_remaining'
-            (mk_error0 (sprintf "Invalid initialization file %s for library\n" cli.input_init))
-      );
+      (* Check for version mismatch. Subtract penalty for mismatch. *)
+      let emsg, rgas = (mk_error0 ("Scilla version mismatch\n")),
+        (Uint64.sub gas_remaining' (Uint64.of_int Gas.version_mismatch_penalty))
+      in
+      let init_json_scilla_version = List.fold_left initargs ~init:None ~f:(fun found (name, lit) ->
+        if is_some found then found else
+        if name = ContractUtil.scilla_version_label
+        then match lit with | UintLit(Uint32L v) -> Some v | _ -> None
+        else None
+      ) in
+      let _ =
+        match init_json_scilla_version with
+        | Some ijv ->
+          let (mver, _, _) = scilla_version in
+          let ijv' = Uint32.to_int ijv in
+          if ijv' <> mver || mver <> lmod.smver
+          then fatal_error_gas emsg rgas
+        | None -> fatal_error_gas emsg rgas
+      in
+
```

Comment:
This is code duplication of https://github.com/Zilliqa/scilla/blob/d9d0ddd8ef34c58b6c851eddb463fba47cdcd73e/src/runners/scilla_runner.ml#L233.
I'd kindly suggest to refactor it.

Reply:
@anton-trunov done :+1: